Backend/Controllers/MedicamentsController.cs

```diff
@@ -55,6 +55,26 @@ public async Task<ActionResult> AddMedicament([FromBody] CreateMedicamentDTO dat
             return Created();
         }
 
+        [HttpPut("{id}")]
+        public async Task<ActionResult> EditMedicament(int id, [FromBody] EditMedicamentDTO dto)
+        {
+            ValidateEditMedicamentDTO(dto);
+            var medicament = await Context.Medicaments.FirstOrDefaultAsync(m => m.Id == id);
+
+            if (medicament == null) return ApiNotFound("Medicament not found!");
+
+            medicament.IsPrescriptionRequired = (bool) dto.IsPrescriptionRequired;
+            medicament.IsReimbursed = (bool) dto.IsReimbursed;
+            medicament.BasePrice = (decimal) dto.BasePrice;
+            medicament.Surcharge = (double) dto.Surcharge;
+            medicament.IsSellable = (bool) dto.IsSellable;
+            medicament.ReimbursePercentage = (int) dto.ReimbursePercentage;
```

Comment:
Maybe we could extract this logic to medicament model and have something like `UpdateMedicamentFromDTO`?

Reply:
fixed